config/mq/config.go

```diff
@@ -17,5 +17,20 @@ func Configure(p *config.Provider) {
 		//  float64 field. Thus here we remove the automatically injected
 		//  cross-resource reference from example manifests.
 		delete(r.References, "configuration.revision")
+
+		r.Sensitive.AdditionalConnectionDetailsFn = func(attr map[string]any) (map[string][]byte, error) {
+			conn := map[string][]byte{}
+			if instances, ok := attr["instances"].([]any); ok && len(instances) > 0 {
+				if instance, ok := instances[0].(map[string]any); ok {
```

Comment:
Hi @loafoe,
Thank you for the PR. Are we not interested in exposing the details of the other instances if `len(instances) > 1` or can there only be at most one instance available here?

Reply:
We have not seen provisions with more than one instance. With that said I was thinking of post fixing the key with the index, just to be future proof. Any suggestion on this?

Reply:
Hi @loafoe,
Yes, agreed, that would be great. We can iterate over the `instances` and export each instance's `console_url` and `enpoint` by post fixing them with the `instances` array index.

Reply:
The connection details secret is part of the MR's public interface so either we can now just use `console_url_0` and `endpoint _0` as connection details secret keys but we do not expose the other items in the `instances` array, or we can implement the for loop and expose all the items in `instances` using secret data keys with the indices.

Reply:
@loafoe How do you feel about @ulucinar's suggestion?

Reply:
@ulucinar @sergenyalcin modified the mapping:

For RabbitMQ:

`instance_0_console_url`
`instance_0_endpoint_0`

For ActiveMQ:

`instance_0_console_url`
`instance_0_endpoint_0`
`instance_0_endpoint_1`
...
`instance_0_endpoint_4`

This captures all data from the Terraform state and allows usage of the broker without a composition.